src/plugins/validation/2and3/semantic-validators/items-required-for-array-objects.js

```diff
@@ -39,11 +40,40 @@ module.exports.validate = function({ jsSpec }) {
       if (Array.isArray(obj.required)) {
         obj.required.forEach((requiredProp, i) => {
           if (!obj.properties || !obj.properties[requiredProp]) {
-            messages.addMessage(
-              path.concat([`required[${i}]`]).join('.'),
-              "Schema properties specified as 'required' must be defined",
-              'error'
-            );
+            // For allOf, ensure atleast one of the child objects has the required property
+            if (obj.allOf) {
+              var props = {};
+              obj.allOf.forEach((childObj) => {
+                childObj = reduceObj(jsSpec, childObj);
+                props = Object.assign(props, childObj.properties);
+              });
+              if (!props[requiredProp]) {
+                messages.addMessage(
+                  path.concat([`required[${i}]`]).join('.'),
+                  "Schema properties specified as 'required' should be defined",
+                  'warning'
```

Comment:
I'm thinking we should make the a configurable check, instead of hard coding a "warning", and maybe setting the default value to error. What do you think?

Reply:
It's a good idea to make it configurable, but i think setting the default value to "warning" would be better since the original problem statement, issue #1920, wanted to reduce importance of this issue from error to warning. @mkistler what do you think?

Reply:
Oh good call - I missed that issue. Warning sounds good to me 👍